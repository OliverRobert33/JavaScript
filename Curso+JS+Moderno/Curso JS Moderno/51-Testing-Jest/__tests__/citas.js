import Citas from '../js/classes/Citas'

describe('Probar la clase de citas', () => {
    const id = Date.now();
    const citas = new Citas();
    test('Agregar a una nueva cita', () => {

        const citaObj = {
            id,
            mascota: 'Pepe',
            propietario: 'OLiver',
            telefono: '0999',
            fecha: '30 abril',
            hora: '13:32',
            sintomas: 'Dolor de cabeza'
        }

        citas.agregarCita(citaObj);


        //Prueba
        expect(citas).toMatchSnapshot()

    });

    test('Actualizar cita', () => {
        const citaObj = {
            id,
            mascota: 'Panchita',
            propietario: 'Roberto',
            telefono: '099249995',
            fecha: '30 abril, 2023',
            hora: '13:34',
            sintomas: 'Dolor de estomago'
        }  

        citas.agregarCita(citaObj)

        //Prueba
        expect(citas).toMatchSnapshot();
    });

    test('EliminaR Cita', () => {
        citas.eliminarCita(id);

        //Prueba
        expect(citas).toMatchSnapshot();
    })
});