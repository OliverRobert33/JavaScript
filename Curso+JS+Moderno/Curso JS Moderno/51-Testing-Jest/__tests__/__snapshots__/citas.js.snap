// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "30 abril",
      "hora": "13:32",
      "id": 1681828915549,
      "mascota": "Pepe",
      "propietario": "OLiver",
      "sintomas": "Dolor de cabeza",
      "telefono": "0999",
    },
    {
      "fecha": "30 abril, 2023",
      "hora": "13:34",
      "id": 1681828915549,
      "mascota": "Panchita",
      "propietario": "Roberto",
      "sintomas": "Dolor de estomago",
      "telefono": "099249995",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar a una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "30 abril",
      "hora": "13:32",
      "id": 1681828915549,
      "mascota": "Pepe",
      "propietario": "OLiver",
      "sintomas": "Dolor de cabeza",
      "telefono": "0999",
    },
  ],
}
`;

exports[`Probar la clase de citas EliminaR Cita 1`] = `
Citas {
  "citas": [],
}
`;
